src/load/azext_load/data_plane/utils/utils.py

```diff
@@ -821,3 +832,62 @@ def validate_engine_data_with_regionwiseload_data(engine_instances, regionwise_e
             f"Sum of engine instances in regionwise load test configuration ({total_engines}) "
             f"should be equal to total engine instances ({engine_instances})"
         )
+
+
+def _get_metrics_from_sampler(test_run, sampler_name, metric_name):
+    return test_run.get("testRunStatistics", {}).get(sampler_name, {}).get(metric_name)
+
+
+def generate_trends_row(test_run, response_time_aggregate=None):
+    trends = {
+        LoadTestTrendsKeys.NAME: test_run.get("testRunId"),
+    }
+    _add_basic_trends(trends, test_run)
+    _add_response_time_trends(trends, test_run, response_time_aggregate)
+    _add_error_and_throughput_trends(trends, test_run)
+    return trends
+
+
+def _add_basic_trends(trends, test_run):
+    if test_run.get("status") is not None:
+        trends[LoadTestTrendsKeys.STATUS] = test_run.get("status")
+    if test_run.get("duration") is not None:
+        trends[LoadTestTrendsKeys.DURATION] = round(test_run.get("duration") / (60 * 1000), 2)
+    if test_run.get("virtualUsers") is not None:
+        trends[LoadTestTrendsKeys.VUSERS] = test_run.get("virtualUsers")
+    sample_count = _get_metrics_from_sampler(test_run, "Total", "sampleCount")
+    if sample_count is not None:
+        trends[LoadTestTrendsKeys.TOTAL_REQUESTS] = sample_count
+
+
+def _add_response_time_trends(trends, test_run, response_time_aggregate):
+    response_time_metrics = {
+        AllowedTrendsResponseTimeAggregations.MEAN.value: "meanResTime",
+        AllowedTrendsResponseTimeAggregations.MEDIAN.value: "medianResTime",
+        AllowedTrendsResponseTimeAggregations.MAX.value: "maxResTime",
+        AllowedTrendsResponseTimeAggregations.MIN.value: "minResTime",
+        AllowedTrendsResponseTimeAggregations.P75.value: "pct75ResTime",
+        AllowedTrendsResponseTimeAggregations.P90.value: "pct1ResTime",
+        AllowedTrendsResponseTimeAggregations.P95.value: "pct2ResTime",
+        AllowedTrendsResponseTimeAggregations.P96.value: "pct96ResTime",
+        AllowedTrendsResponseTimeAggregations.P98.value: "pct98ResTime",
+        AllowedTrendsResponseTimeAggregations.P99.value: "pct3ResTime",
+        AllowedTrendsResponseTimeAggregations.P999.value: "pct999ResTime",
+        AllowedTrendsResponseTimeAggregations.P9999.value: "pct9999ResTime",
+    }
+    for key, metric in response_time_metrics.items():
+        if response_time_aggregate == key:
+            if os.getenv("AZDEV_TEST_ENV"):
+                trends[getattr(LoadTestTrendsKeys, key.upper() + "_RES_TIME")] = None
```

Comment:
just some handling for unit tests, since the data would not actually be available, but for sanity, the key would be. In prod, the key wouldn't be available in response if corresponding data isn't present

Reply:
Can we not get the key for tests as well? I dont prefer putting test specific handling in main code, unless absolutely needed?

Reply:
naa, not needed, keeping for prod too